tools/utils.py

```diff
@@ -93,3 +93,16 @@ def write_binary(file_path, contents):
   """Write to a file opened in binary mode"""
   with open(file_path, 'wb') as fh:
     fh.write(contents)
+
+
+# Verify the the packages from `requirements.txt` are installed correctly.
+# Emsdk using `emscripten/.pip` when installing these packages so add this to the
+# system path first.
+pip_path = os.path.join(path_from_root(), '.pip')
+if os.path.exists(pip_path):
+  sys.path.append(pip_path)
+
+try:
+  import leb128  # noqa
+except ImportError:
+  exit_with_error('unable to import `leb128`.  Did you run `pip install -r requirements.txt`?"')
```

Comment:
It looks like this is checking if pip is installed, even if nothing is really used from a pip package?

I think it would be good to not do this, but instead do it at actual sites where such a package is attempted to be used (i.e. like we handle closure compiler).

That way people don't need to install pip when they don't really need anything from there. Currently for example looks like tools/webassembly.py is the only consumer of leb128, and nothing uses tools/webassembly.py itself, it is just a tool provided for end users?

So this would be asking people to unconditionally do pip install even though they don't need anything from there?

Reply:
The webassembly.py is used in several places now, and most recently to extra the metadata from the wasm file (eliminating the dependency on wasm-emscripten-finalize) : https://github.com/emscripten-core/emscripten/pull/16529.   Its not a standalone tool but a core part of the compiler .. not optional.